### ISSUE TITLE
Template options

### ISSUE THREAD

**Issue by jmeas** [marionettejs/backbone.marionette#2165]
Right now, our renderer doesn't accept [template options](https://github.com/marionettejs/backbone.marionette/blob/master/src/template-cache.js#L89-L95). Why not?

I've implemented global and per-instance options in Puppets [here.](https://github.com/jmeas/puppets/blob/v0.0.1/src/utilities.js#L16-L28)

**Comment by jmeas** [comment]
>  There shouldn't be global settings... that's really crappy.

This sentiment doesn't make any sense to me. The Renderer and TemplateCache objects *themselves* are global settings. All that we're doing is making our global settings more flexible.

**Comment by thejameskyle** [comment]
@jmeas I'm saying that I was these to be prototype-based. I don't want a global TemplateCache or Renderer that I have no option of localizing, which is exactly why I want it to exist in the View class.

ie. 

```js
class VDOMView extends View {
  compile(template) {
    return VDOM.compile(template);
  }
}

class HandlebarsView extends View {
  compile(template) {
    return Handlebars.compile(template);
  }
}
```

**Comment by jasonLaster** [comment]
Yea, I think you're right @thejameskyle

**Comment by thejameskyle** [comment]
If people do want to change a global setting, they should do it the way other things are done.

```js
View.prototype.compile = VDOM.compile;
```

All View classes should extend from one common place, and if setting something like this up isn't as straightforward as overriding 1 or 2 methods on `View.prototype`, then there's something wrong with it. (VDOM is likely more complicated though).

**Comment by jmeas** [comment]
I tried arguing for that once before, @thejameskyle. @samccone was against it

**Comment by thejameskyle** [comment]
@samccone can you explain your opposition?

**Comment by jasonLaster** [comment]
So I guess in this world if one Layout view is vdom and another is HBs, you could extend layoutView twice... Not too bad

**Comment by thejameskyle** [comment]
I see people using multiple types of templating/rendering engines more and more, especially with React.

**Comment by samccone** [comment]
yeah this got merged#2205
